### PR TITLE
Set up search for selenium/integration tests

### PR DESCRIPTION
**Pull request by mwiencek**
This will allow us to write tests for pages/features that make search queries.

### CODE REVIEW

**Comment by mwiencek** [comment]
I removed the bits from our selenium tests that click "switch to direct search" in the autocompletes to show that this works, but I'll also update this with an additional test for the release editor's duplicate selection tab.

**Comment by mwiencek** [comment]
Fixed conflicts & improved the coverage a bit on the new duplicates tab test.

File: t/sql/selenium.sql [hunk 1]
```diff
@@ -55,9 +55,14 @@ INSERT INTO event (id, gid, name, begin_date_year, begin_date_month, begin_date_
...
     ('a428f34a-9761-3544-2279-a58cbf15ba9d', 1606, '2014-12-18 13:00:34.106501+00');
 
+DROP TRIGGER a_ins_instrument ON instrument;
+
 INSERT INTO instrument (id, gid, name, type, edits_pending, last_updated, comment, description) VALUES
     (55, '7ee8ebf5-3aed-4fc8-8004-49f4a8c45a87', 'electric guitar', 2, 0, '2014-08-22 17:09:51.69681+00', '', '');
 
+CREATE TRIGGER a_ins_instrument AFTER INSERT ON instrument
+    FOR EACH ROW EXECUTE PROCEDURE a_ins_instrument();
+
```
[the remark this answers is not in the document]
Comment (mwiencek) [reply t/sql/selenium.sql]: I must've added it at some point to prevent it from adding instruments to the link_attribute_type table, but whatever I needed that for it doesn't seem to be necessary anymore, so removed it

**Comment by mwiencek** [comment]
@yvanzo thanks, I applied all your suggested improvements :)

File: docker/Dockerfile.tests [hunk 2]
```diff
@@ -58,27 +57,78 @@ RUN apt-get update && \
...
+        maven \
+        openjdk-8-jdk \
+        openjdk-8-jre \
         openssh-client \
         perl \
         postgresql \
         postgresql-10-pgtap \
         postgresql-contrib \
         postgresql-server-dev-10 \
+        python-dev \
```
Comment (yvanzo) [inline docker/Dockerfile.tests]: `pip2` is needed to install SIR v1.0.2

```suggestion
        python2.7-dev \
```

File: docker/Dockerfile.tests [hunk 3]
```diff
@@ -58,27 +57,77 @@ RUN apt-get update && \
...
+    mkdir -p /opt/solr/lib $SOLR_HOME && \
+    cp target/mb-solr-0.0.1-SNAPSHOT-jar-with-dependencies.jar /opt/solr/lib/ && \
+    cd .. && \
+    cp -R mbsssss $SOLR_HOME/mycores/ && \
+    sed -i'' 's|</solr>|<str name="sharedLib">/opt/solr/lib</str></solr>|' $SOLR_HOME/solr.xml && \
+    mkdir $SOLR_HOME/data && \
+    chown -R solr:solr /opt/solr/ && \
+    cd /home/musicbrainz
+
+RUN sudo -E -H -u musicbrainz git clone https://github.com/metabrainz/sir.git && \
```
[the remark this answers is not in the document]
Comment (mwiencek) [reply docker/Dockerfile.tests]: This broke because https://github.com/metabrainz/sir/commit/7c4c50b27367d4c2e2e2a29626801b12a9d4d388 isn't included in that tag. I'll revert this but add a line to hard reset the commit.

**Comment by mwiencek** [comment]
> All python -m sir calls should be made from the virtual environment, including those from script/reset_selenium_env.sh.

That script runs `. venv/bin/activate` on line 65 already, so I think it's fine.

File: docker/Dockerfile.tests (hunk 2 above)
Comment (mwiencek) [reply docker/Dockerfile.tests]: I changed this, though I'm not sure what it does - looking at https://packages.ubuntu.com/bionic/python-dev (phusion/baseimage:0.11 uses 18.04), it was for python2 and python2.7-dev was a dependency of it.

File: docker/Dockerfile.tests [hunk 4]
```diff
@@ -58,28 +57,80 @@ RUN apt-get update && \
...
+    sed -i'' 's|</solr>|<str name="sharedLib">/opt/solr/lib</str></solr>|' $SOLR_HOME/solr.xml && \
+    mkdir $SOLR_HOME/data && \
+    chown -R solr:solr /opt/solr/ && \
+    cd /home/musicbrainz
+
+ENV SIR_COMMIT 7c4c50b
+
+RUN sudo -E -H -u musicbrainz git clone https://github.com/metabrainz/sir.git && \
+    cd sir && \
+    git reset --hard $SIR_COMMIT && \
```
Comment (yvanzo) [inline docker/Dockerfile.tests]: Doesn’t it require sudo too?

```suggestion
    sudo -E -H -u git reset --hard $SIR_COMMIT && \
```

Comment (mwiencek) [reply docker/Dockerfile.tests]: I guess not since it works without it, but that would be more correct & avoid any permission issues in the future.

**Comment by mwiencek** [comment]
Rebased & pointed the CircleCI config to a new image built with the `sudo` change, just for good measure.
